intercom-common/src/foreign_ty.rs

```diff
@@ -66,23 +181,182 @@ impl ForeignTypeHandler for CTypeHandler
+            Some( _ ) => panic!("Cannot set pass_by twice."),
+            None => {},
+        }
+
+        TypeInfoResolver {
+            krate: resolver.krate,
+            rust_type: resolver.rust_type,
+            pass_by: Some( pass_by ),
+            is_mutable: resolver.is_mutable,
+            array_length: resolver.array_length,
+        }
+    }
+
+    fn mutable_option(
+        resolver: Option<TypeInfoResolver<'s>>,
+        is_mutable: bool,
+    ) -> Option<TypeInfoResolver<'s>>
+    {
+        match resolver {
+            Some( r ) => Some( TypeInfoResolver::mutable( r, is_mutable ) ),
+            None => None,
+        }
+    }
+
+    fn mutable(
+        resolver: TypeInfoResolver<'s>,
+        is_mutable: bool,
+    ) -> TypeInfoResolver<'s>
+    {
+        match resolver.is_mutable {
+            Some(_) => panic!("Cannot set is_mutable twice."),
+            None => {}
+        }
+
+        TypeInfoResolver {
+            krate: resolver.krate,
+            rust_type: resolver.rust_type,
+            pass_by: resolver.pass_by,
+            is_mutable: Some( is_mutable ),
+            array_length: resolver.array_length,
+        }
+    }
+
+    fn array_option(
+        resolver: Option<TypeInfoResolver<'s>>,
+        array_length: &'p syn::Expr,
+    ) -> Option<TypeInfoResolver<'s>>
+    {
+        match resolver {
+            Some( r ) => Some( TypeInfoResolver::array( r, array_length ) ),
+            None => None,
+        }
+    }
+
+    fn array(
+        resolver: TypeInfoResolver<'s>,
+        array_length: &'p syn::Expr,
+    ) -> TypeInfoResolver<'s>
+    {
+        match resolver.array_length {
+            Some(_) => panic!("Cannot set array_length twice."),
+            None => {}
+        }
+
+        TypeInfoResolver {
+            krate: resolver.krate,
+            rust_type: resolver.rust_type,
+            pass_by: resolver.pass_by,
+            is_mutable: resolver.is_mutable,
+            array_length: Some( array_length ),
+        }
+    }
+
+    fn from_array(
+        krate : &'p ComCrate,
+        array: &'p syn::TypeArray,
+    ) -> Option<TypeInfoResolver<'s>>
+    {
+        TypeInfoResolver::array_option(
+            TypeInfoResolver::from_type( krate, &array.elem ),
+            &array.len,
+        )
+    }
+
+    fn from_slice(
+        krate : &'p ComCrate,
+        slice: &'p syn::TypeSlice,
+    ) -> Option<TypeInfoResolver<'s>>
+    {
+        TypeInfoResolver::from_type( krate, &slice.elem )
+    }
+
+    fn from_reference(
+        krate : &'p ComCrate,
+        reference : &'p syn::TypeReference,
+    ) -> Option<TypeInfoResolver<'s>>
+    {
+        TypeInfoResolver::pass_by_option(
+                TypeInfoResolver::mutable_option(
+                            TypeInfoResolver::from_type( krate, &reference.elem ),
+                            TypeInfoResolver::is_mutable( &reference.mutability )
+                        ),
+                PassBy::Reference,
+        )
```

Comment:
The `pass_by_option` just calls `pass_by` if the first parameter is `Some` and otherwise returns `None`

Couldn't that call be eliminated by just going through:

```rust
TypeInfoResolver::pass_by(
    TypeInfoResolver::mutable_option( ... )?,
    //                                     ^
    //   If 'mutable_option' returns None, ? will exit early with 'None'. Otherwise it unwraps the Some value.
    PassBy::Reference
)

Reply:
Same applies to the various other `_option` methods.